solver/src/solver/http_solver.rs

```diff
@@ -264,12 +270,36 @@ impl HttpSolver {
     ) -> Result<(BatchAuctionModel, SettlementContext)> {
         let tokens = self.map_tokens_for_solver(liquidity.as_slice());
 
-        let (token_infos, price_estimates) = join!(
+        let (token_infos, price_estimates, mut buffers_result) = join!(
             self.token_info_fetcher.get_token_infos(tokens.as_slice()),
             self.price_estimator
-                .estimate_prices(tokens.as_slice(), self.native_token)
+                .estimate_prices(tokens.as_slice(), self.native_token),
+            self.buffer_retriever.get_buffers(tokens.as_slice())
         );
 
+        let _buffers: HashMap<_, _> = buffers_result
+            .drain()
+            .filter_map(|(token, buffer)| match buffer {
+                Err(BufferRetrievalError::Erc20(err)) if is_transaction_failure(&err.inner) => {
```

Comment:
Is there a reason you prefer to handle this here instead of in `BufferRetrieving::get_buffers`?

Reply:
It was a change based on [this comment](https://github.com/gnosis/gp-v2-services/pull/934#discussion_r681803290). I personally don't mind either way where the error is handled.